### ISSUE TITLE
No rule is validating If I don't specify required rule first 

### ISSUE THREAD

**Issue by xsaad67** [vlucas/valitron#226]
Hello, I'm facing a quite weird issue. When I start validating my form If I don't specify the required rule. no validation just work at all. Just look at the below snippet

```
$v = new Valitron\Validator($_POST);
$v->rule('email','email');
if($v->validate()){
		echo "yes";
	}else{
		print_r($v->errors());
	}
```

As I suppose the above code will validate email so a valid email can't be empty? But still for Validator a empty email is ok. Now just look at the below code

```
$v = new Valitron\Validator($_POST);
$v->rule('required', array('email'))->message('{field} is required');
$v->rule('email','email');
$v->labels(array(
    'email' => 'Email Address',
));
```
After adding a required in all of sudden email become an invalid and I got two errors one for **required rule** and the other one is for **email rule**

My question is # How come an empty email can be a valid email?

**Comment by willemwollebrants** [comment]
Valitron only validates fields that are not empty, unless they are also required. So when you pass an empty string as the email address, Valitron first checks if the field has a 'required'-rule. It hasn't, so the rules for that field are ignored. 

Without the required rule, what you have is: "if there is a value in this field, it must be an e-mail address". With the required rule this becomes: "there must be a value in this field, and it must be an e-mail address".

So in short, if you must have an e-mail address, the field must also be required.

**Comment by atishagrawal** [comment]
I think @xsaad67 had the same question. He wanted to suggest an option to eliminate **required** rule in cases where there is another validation rule already passed. I agree with @willemwollebrants regarding his opinion about that an email-address being validated must be required, but don't you think that do developers explicitly need to mention it?

However, in cases, where email is not mandatory (is optional), then I think **required** rule doesn't suit. But if a user has entered his/her email, then a valid email is necessary. In such cases, the developer doesn't have to make it required.

**Comment by Bodom78** [comment]
I must have missed this in the instructions and got caught out by it as well.

Wouldnt it be better to have all assigned rules run, I mean custom rules could check if the data exists and how to action right?

**Comment by willemwollebrants** [comment]
It always comes down to the same thing: if a field is required, it must be marked as required. If a field is not required, this means that an empty value is acceptable. If an empty value is not acceptable, the field is required.
